### PR TITLE
[AIRFLOW-1397] No Last Run column data displyed in Airflow UI 1.8.1

### PR DESCRIPTION
**Pull request by ghost**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1397
      "Airflow 1.8.1 - No data displays in Last Run Column in Airflow UI"

### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:
Associated Jira issue: https://issues.apache.org/jira/browse/AIRFLOW-1397
Last Run was not displaying Airflow UI version 1.8.1, this pr will display them.

Screenshot before the change
screen shot 2017-07-11 at 12 24 54 pm

Screenshot after the change
screen shot 2017-07-11 at 12 26 31 pm

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
These changes are related to Airflow UI.

### Commits
[AIRFLOW-1397] No Last Run column data displyed in Airflow UI 1.8.1

### CODE REVIEW

**Comment by bolkedebruin** [comment]
Can you please correct your commit message ([AIRFLOW-XXX]) and provide screenshot of before and after? Thanks!

File: airflow/models.py [hunk 1]
```diff
@@ -4359,12 +4359,13 @@ def is_backfill(self):
     @provide_session
     def get_latest_runs(cls, session):
         """Returns the latest running DagRun for each DAG. """
+        dag_states=[State.RUNNING, State.SUCCESS, State.FAILED]
```
Comment (bolkedebruin) [inline airflow/models.py]: Use State.finished()

Comment (ghost) [reply airflow/models.py]: thanks, corrected it.

**Comment by saguziel** [comment]
To be clear, you should have square brackets around AIRFLOW-1397 so it should  be `[AIRFLOW-1397]`

**Comment by saguziel** [comment]
I don't think this is the right logic nor the right fix

**Comment by saguziel** [comment]
Now the logic is completely different from the comments and the old behavior. Of course, we can change it, but we'd have to update a bunch of stuff.

File: airflow/models.py (hunk 1 above)
Comment (ghost) [reply airflow/models.py]: i have tried correcting the commit message as well, please review my pr once again and let me know if it needs any more changes. thanks.

**Review by ghost — COMMENTED** [review]
@bolkedebruin would you mind taking a second look into my pr and let me know if it needs any more changes? https://github.com/apache/incubator-airflow/pull/2430

**Comment by bolkedebruin** [comment]
@saguziel care to elaborate? The change seems like a bug fix to me, but maybe I am wrong? It is also just visual thing so I don't understand your 'we need to update a bunch of stuff'.

**Comment by saguziel** [comment]
You're changing the behavior of the endpoint, and at this point it doesn't match the docstring, and in fact, does the opposite.

**Comment by ghost** [comment]
@saguziel i didn't get when you said it does the opposite. This bug fix shows the last run values of dag that is missing from the current version. Can you please give me some screenshots of code or UI that can elaborate more?

**Comment by saguziel** [comment]
The behavior doesn't match with the docstring, nor with any of the upstreams (see also latest_dag_runs in api.experimental)

**Comment by saguziel** [comment]
It's not really a bug fix when the behavior of that column was that it would show the execution_date of the latest running dag_run.

**Comment by saguziel** [comment]
But perhaps the documentation doesn't make it clear. We can go that route, but either way, this change on its own is not complete.

**Comment by saguziel** [comment]
IMO, it would probably sense to remove the condition filtering on dagrun state, and update all the documentation

**Comment by Acehaidrey** [comment]
Is there anything else we're waiting for on this pr? It's affecting our workflow as well and would be great to have this merged into master :)

**Comment by saguziel** [comment]
I think the tests need to be updated as well

File: airflow/models.py [hunk 2]
```diff
@@ -4358,13 +4358,13 @@ def is_backfill(self):
...
     def get_latest_runs(cls, session):
-        """Returns the latest running DagRun for each DAG. """
+        """Returns the latest DagRun for each DAG. """
         subquery = (
             session
             .query(
                 cls.dag_id,
                 func.max(cls.execution_date).label('execution_date'))
-            .filter(cls.state == State.RUNNING)
+            .filter(cls.state.in_(State.finished()))
```
Comment (saguziel) [inline airflow/models.py]: I don't think we should filter on finished. Even if we decide to, the documentation should match

Comment (saguziel) [reply airflow/models.py]: I'm suggesting just deleting this line entirely

**Comment by Acehaidrey** [comment]
This fix would be great for me as well. Thanks @preete-dixit-ck for working on this. Is there anything else this ticket is waiting on? Looking forward to pulling this change

**Comment by ghost** [comment]
@Acehaidrey thanks for your feedback. I understand as we also make this code change manually everytime we install airflow on new machine. I am not sure how the process works here to merge the pull requests. i guess @bolkedebruin can help us here to go forward here.

File: airflow/models.py [hunk 3]
```diff
@@ -4358,13 +4358,13 @@ def is_backfill(self):
...
     def get_latest_runs(cls, session):
-        """Returns the latest running DagRun for each DAG. """
+        """Returns the latest DagRun for each DAG. """
         subquery = (
             session
             .query(
                 cls.dag_id,
                 func.max(cls.execution_date).label('execution_date'))
-            .filter(cls.state == State.RUNNING)
+                .filter(cls.state.in_(State.finished()))
```
Comment (saguziel) [inline airflow/models.py]: Delete this line

**Comment by saguziel** [comment]
You need to delete that line, you re-added it.

**Comment by Acehaidrey** [comment]
Also I think they'll want you to squash your commits to be just one. https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git.

**Comment by saguziel** [comment]
Please squash your commits

**Comment by ghost** [comment]
@saguziel squashed all commit into single one, please check.

**Comment by saguziel** [comment]
@preete-dixit-ck I'd recommend adding a fix version to your commit

**Pull request closed without merging**